test/miscellaneous.jl

```diff
@@ -680,30 +680,46 @@ using Test
         function smoothdim_avx!(s, x, α, Rpre, irng::AbstractUnitRange, Rpost)
             ifirst, ilast = first(irng), last(irng)
             ifirst > ilast && return s
-            @avx for Ipost in Rpost
-                # Initialize the first value along the filtered dimension
+            @avx tile=(1,1) for Ipost in Rpost
+                # Handle all other entries
                 for Ipre in Rpre
                     s[Ipre, ifirst, Ipost] = x[Ipre, ifirst, Ipost]
+                    for i = ifirst+1:ilast
+                        s[Ipre, i, Ipost] = α*x[Ipre, i, Ipost] + (1-α)*x[Ipre, i-1, Ipost]
+                    end
                 end
+            end
+            s
+        end
+        function smoothdim_ifelse_avx!(s, x, α, Rpre, irng::AbstractUnitRange, Rpost)
+            ifirst, ilast = first(irng), last(irng)
+            ifirst > ilast && return s
+            @avx tile=(1,1) for Ipost in Rpost
                 # Handle all other entries
-                for i = ifirst+1:ilast
+                for i = ifirst:ilast
                     for Ipre in Rpre
-                        s[Ipre, i, Ipost] = α*x[Ipre, i, Ipost] + (1-α)*x[Ipre, i-1, Ipost]
+                        xi = x[Ipre, i, Ipost]
+                        xim = ifelse(i == ifirst, xi, x[Ipre, i-1, Ipost])
+                        s[Ipre, i, Ipost] = α*xi + (1-α)*xim
                     end
                 end
```

Comment:
Not as pretty, but another way to write the function. 

I think this may risk crashing because of the read to `x[Ipre, ifirst-1, Ipost]`, even though that value wont be used, so I'm not inclined to recommend this until I add explicit support for conditional loads, like the conditional store support it has now.

Reply:
This did cause crashes on Travis. I'll work on turning this into a conditional load.